src/connection.js

```diff
@@ -111,6 +111,8 @@ export class StableWSConnection {
 			},
 		);
 
+		this.wsID += 1;
```

Comment:
Repeating the comment (since last one got removed due to updated commit)
===================================

This is a disconnect function, which one uses when he wants to manually disconnect the websocket connection e.g. when mobile app goes to background.

For some weird reason, ws hits the original onclose (Line 357) callback (with code 1001) after ws.close() (line 159), which should not happen ideally since we override onclose callback on line 140. This is NOT a reproducible issue on our side. But some customer is facing this issue. The logs that he provided didn't help figure out why is this happening.

The main problem is - when original onclose callback gets called, it triggers the reconnection logic ... which defeats the whole purpose manually disconnecting the client. The moment user calls disconnect, our code immediately triggers reconnect.

To avoid this, instead of reseting the this.wsID to 1, we can increment it, so that original callback function will not respond anymore( line 364)

ZD ticket - https://getstream.zendesk.com/agent/tickets/1805